app/models/assignment.rb

```diff
@@ -1094,53 +1094,42 @@ def create_peer_review_assignment_if_not_exist
 
   ### REPO ###
 
-  def self.repository_names
-    pluck(:short_identifier).map { |sid| STARTER_CODE_REPO_FORMAT % sid }
-  end
-
-  def repository_name
-    STARTER_CODE_REPO_FORMAT % short_identifier
-  end
-
-  def build_repository
-    # create repositories if and only if we are admin
-    return true unless MarkusConfigurator.markus_config_repository_admin?
-    # only create if we can add starter code
-    return true unless can_upload_starter_code?
+  def build_starter_code_repo
+    return unless MarkusConfigurator.markus_config_repository_admin?
+    return unless can_upload_starter_code?
+    repo_dir = File.join(MarkusConfigurator.markus_config_repository_storage, STARTER_CODE_REPO_NAME)
     begin
-      Repository.get_class.create(File.join(MarkusConfigurator.markus_config_repository_storage, repository_name))
-    rescue Repository::RepositoryCollision => e
-      # log the collision
-      errors.add(:base, self.repository_name)
+      unless Repository.get_class.repository_exists?(repo_dir)
+        Repository.get_class.create(repo_dir, with_hooks: false)
+      end
+      access_starter_code_repo do |repo|
+        txn = repo.get_transaction('Markus', I18n.t('repo.commits.starter_code', assignment: self.short_identifier))
+        txn.add_path(self.repository_folder)
+        repo.commit(txn)
+      end
+    rescue StandardError => e
+      errors.add(:base, repo_dir)
       m_logger = MarkusLogger.instance
-      m_logger.log("Creating repository '#{repository_name}' caused repository collision. " +
-                     "Error message: '#{e.message}'",
-                   MarkusLogger::ERROR)
+      m_logger.log("Creating repository '#{repo_dir}' failed with '#{e.message}'", MarkusLogger::ERROR)
     end
-    access_repo do |repo|
-      txn = repo.get_transaction('Markus')
-      txn.add_path(repository_folder)
-      repo.commit(txn)
-    end
-    true
   end
 
-  def repo_loc
-    repo_loc = File.join(MarkusConfigurator.markus_config_repository_storage, repository_name)
+  def starter_code_repo_path
+    repo_loc = File.join(MarkusConfigurator.markus_config_repository_storage, STARTER_CODE_REPO_NAME)
     unless Repository.get_class.repository_exists?(repo_loc)
```

Comment:
Since you're changing this method, I'd prefer putting the `exists?` check into `starter_code_repo` and `access_starter_code_repo` to keep this method pure. It would allow you to use the method up in `build_starter_code_repo`, for example.